File: BACKEND/src/main/java/com/airbnb/clone/User/Service/UserService.java
package com.airbnb.clone.User.Service;



import com.airbnb.clone.User.UserException;
import com.airbnb.clone.User.Entity.User;


public interface UserService {

	User Register(User User) throws UserException;
	User Login(User User);
	boolean DeleteUser(Long Id);
	User Update(Long Id,User User) throws UserException;
	User getUserByEmail(String Email);
	User getUserById(Long Id) throws UserException;
	User becomeHost(Long userId) throws UserException;
}

File: FRONTEND/src/services/UserService.js
import axios from "axios";

const API_URL = "http://localhost:8086/user";
const AUTH_URL = "http://localhost:8086/auth";

export const getUserById = (id) => {
  return axios.get(`${API_URL}/getUser/${id}`);
};

export const getUserByEmail = (email) => {
  return axios.get(`${API_URL}/getByEmail/${email}`);
};

export const registerUser = (userData) => {
  return axios.post(`${API_URL}/public/register`, userData);
};

export const updateUser = (id, userData) => {
  return axios.put(`${API_URL}/updateUser/${id}`, userData);
};

export const loginUser = (email, password) => {
  return axios.post(`${AUTH_URL}/login`, { email, password });
};

export const becomeHost = (userId) => {
  return axios.post(`${API_URL}/become-host/${userId}`);
};
